### ISSUE TITLE
Infinite bias value with Logistic Regression algorithm

### ISSUE THREAD

**Issue by qf6101** [Angel-ML/angel#316]
I tried running the new logistic regression algorithm in classification2 directory. After the first epoch is finished, the metric info collection process leads to the following error. The real string is NaN. And I found the value of intercept is '-infinite' in classification2.LRLearner.validate method when invoking LRModel.calPredAndLoss method and more step into the line: val bias = getBias(params.get(intercept)).

The returned localParams in  'val localParams = trainOneEpoch(epoch, trainData, indexes)' is invalid.
```
18/04/17 13:57:37 FATAL utils.AngelRunJar: submit job failed 
java.lang.NumberFormatException: For input string: "�"
        at sun.misc.FloatingDecimal.readJavaFormatString(FloatingDecimal.java:2043)
        at sun.misc.FloatingDecimal.parseDouble(FloatingDecimal.java:110)
        at java.lang.Double.parseDouble(Double.java:538)
        at java.lang.Double.valueOf(Double.java:502)
        at com.tencent.angel.client.AngelClient.toString(AngelClient.java:535)
        at com.tencent.angel.client.AngelClient.updateJobReport(AngelClient.java:523)
        at com.tencent.angel.client.AngelClient.isExecuteCompleted(AngelClient.java:459)
        at com.tencent.angel.client.AngelClient.waitForCompletion(AngelClient.java:338)
        at com.tencent.angel.ml.MLRunner$class.train(MLRunner.scala:50)
        at com.tencent.angel.ml.classification2.lr.LRRunner.train(LRRunner.scala:30)
        at com.tencent.angel.ml.classification2.lr.LRRunner.train(LRRunner.scala:41)
        at com.tencent.angel.ml.MLRunner$class.submit(MLRunner.scala:93)
        at com.tencent.angel.ml.classification2.lr.LRRunner.submit(LRRunner.scala:30)
        at com.tencent.angel.utils.AngelRunJar$1.run(AngelRunJar.java:67)
        at com.tencent.angel.utils.AngelRunJar$1.run(AngelRunJar.java:54)
        at java.security.AccessController.doPrivileged(Native Method)
        at javax.security.auth.Subject.doAs(Subject.java:422)
        at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1698)
        at com.tencent.angel.utils.AngelRunJar.submit(AngelRunJar.java:54)
        at com.tencent.angel.utils.AngelRunJar.main(AngelRunJar.java:38)
```

The following are some logs.
```
2018-04-18 14:42:09,346 INFO [pool-6-thread-1] com.tencent.angel.ml.optimizer2.OptModel: Start to pull lr_intercept from PS ...
2018-04-18 14:42:09,346 INFO [pool-6-thread-1] com.tencent.angel.psagent.consistency.ConsistencyController: wait for clock 19
2018-04-18 14:42:09,347 INFO [pool-6-thread-1] com.tencent.angel.ml.optimizer2.OptModel: lr_intercept is pulled !
2018-04-18 14:42:10,849 INFO [pool-6-thread-1] com.tencent.angel.ml.optimizer2.Optimizer: Process: 100% of samples have finished ! The current average loss is 0.2533909536054331
2018-04-18 14:42:10,853 INFO [pool-6-thread-1] com.tencent.angel.ml.optimizer2.OptModel: Start to push lr_weight from PS ...
2018-04-18 14:42:10,931 INFO [pool-6-thread-1] com.tencent.angel.ml.optimizer2.OptModel: lr_weight is pushed !
2018-04-18 14:42:10,931 INFO [pool-6-thread-1] com.tencent.angel.ml.optimizer2.OptModel: Start to push lr_intercept from PS ...
2018-04-18 14:42:10,941 INFO [pool-6-thread-1] com.tencent.angel.ml.optimizer2.OptModel: lr_intercept is pushed !
2018-04-18 14:42:11,012 INFO [pool-6-thread-1] com.tencent.angel.ml.optimizer2.OptModel: Start to pull lr_weight from PS ...
2018-04-18 14:42:11,012 INFO [pool-6-thread-1] com.tencent.angel.psagent.consistency.ConsistencyController: wait for clock 21
2018-04-18 14:42:14,349 INFO [pool-6-thread-1] com.tencent.angel.ml.optimizer2.OptModel: lr_weight is pulled !
2018-04-18 14:42:14,349 INFO [pool-6-thread-1] com.tencent.angel.ml.optimizer2.OptModel: Start to pull lr_intercept from PS ...
2018-04-18 14:42:14,350 INFO [pool-6-thread-1] com.tencent.angel.psagent.consistency.ConsistencyController: wait for clock 21
2018-04-18 14:42:14,350 INFO [pool-6-thread-1] com.tencent.angel.ml.optimizer2.OptModel: lr_intercept is pulled !
2018-04-18 14:42:14,366 INFO [pool-6-thread-1] com.tencent.angel.ml.optimizer2.Optimizer: The nonZeroNumber ratio is 1.0
2018-04-18 14:42:14,366 INFO [pool-6-thread-1] com.tencent.angel.ml.optimizer2.Optimizer: The pull, calculate, push time is this epoch are 115358, 15186 and 1285 respectively !
2018-04-18 14:42:19,558 INFO [pool-6-thread-1] com.tencent.angel.worker.storage.DiskDataBlock: shift to read next file: aaf39df086e4_WorkerAttempt_15_15_0_15_1
2018-04-18 14:42:23,954 INFO [pool-6-thread-1] com.tencent.angel.ml.classification2.lr.LRLearner: Task[15]: epoch = 0 trainData loss = NaN precision = 0.14611088826345414 auc = 0.005726684385471607 trueRecall = 0.0 falseRecall = 1.0
2018-04-18 14:42:23,955 INFO [pool-6-thread-1] com.tencent.angel.ml.classification2.lr.LRLearner: Task[15]: epoch = 0 validationData loss = NaN precision = 0.07692307692307693 auc = 0.0 trueRecall = 0.0 falseRecall = 1.0
2018-04-18 14:42:23,955 INFO [pool-6-thread-1] com.tencent.angel.ml.classification2.lr.LRLearner: Task[15]: epoch=0 success. epoch cost 141434 ms.train cost 131845 ms. validation cost 9589 ms.
2018-04-18 14:42:23,967 INFO [pool-6-thread-1] com.tencent.angel.ml.classification2.lr.LRLearner: Task[15]: epoch=1 start.
2018-04-18 14:42:23,967 INFO [pool-6-thread-1] com.tencent.angel.ml.optimizer2.OptModel: Start to pull lr_weight from PS ...
2018-04-18 14:42:23,968 INFO [pool-6-thread-1] com.tencent.angel.psagent.consistency.ConsistencyController: wait for clock 21
2018-04-18 14:42:23,985 INFO [pool-6-thread-1] com.tencent.angel.ml.optimizer2.OptModel: lr_weight is pulled !
2018-04-18 14:42:23,985 INFO [pool-6-thread-1] com.tencent.angel.ml.optimizer2.OptModel: Start to pull lr_intercept from PS ...
2018-04-18 14:42:23,985 INFO [pool-6-thread-1] com.tencent.angel.psagent.consistency.ConsistencyController: wait for clock 21
2018-04-18 14:42:23,986 INFO [pool-6-thread-1] com.tencent.angel.ml.optimizer2.OptModel: lr_intercept is pulled !
2018-04-18 14:42:25,212 INFO [pool-6-thread-1] com.tencent.angel.ml.optimizer2.Optimizer: Process: 10% of samples have finished ! The current average loss is NaN
2018-04-18 14:42:25,217 INFO [pool-6-thread-1] com.tencent.angel.ml.optimizer2.OptModel: Start to push lr_weight from PS ...
2018-04-18 14:42:25,316 INFO [pool-6-thread-1] com.tencent.angel.ml.optimizer2.OptModel: lr_weight is pushed !
2018-04-18 14:42:25,316 INFO [pool-6-thread-1] com.tencent.angel.ml.optimizer2.OptModel: Start to push lr_intercept from PS ...
2018-04-18 14:42:25,326 INFO [pool-6-thread-1] com.tencent.angel.ml.optimizer2.OptModel: lr_intercept is pushed !
2018-04-18 14:42:25,398 INFO [pool-6-thread-1] com.tencent.angel.ml.optimizer2.OptModel: Start to pull lr_weight from PS ...
2018-04-18 14:42:25,398 INFO [pool-6-thread-1] com.tencent.angel.psagent.consistency.ConsistencyController: wait for clock 23
```

The following is the command.
```
angel-1.5.0-bin/bin/angel-submit \
     --action.type train \
     --angel.app.submit.class com.tencent.angel.ml.classification2.lr.LRRunner  \
     --angel.train.data.path $input_path \
     --angel.save.model.path $model_path \
     --angel.log.path $logpath \
     --ml.epoch.num 100 \
     --ml.num.update.per.epoch 10 \
     --ml.minibatch.size 128 \
     --ml.data.validate.ratio 0.000001 \
     --ml.feature.index.range 879248 \
     --ml.batch.sample.ratio 1.0 \
     --ml.data.input.type libsvm \
     --ml.learn.rate 0.001 \
     --ml.learn.decay 0.1 \
     --ml.lr.reg.l1 0.1 \
     --ml.lr.reg.l2 0.1 \
     --angel.workergroup.number 400 \
     --angel.worker.memory.gb 10 \
     --angel.worker.task.number 1 \
     --angel.ps.number 100 \
     --angel.ps.memory.gb 5 \
     --angel.job.name angel_lr_data
```

**Comment by wangcaihua** [comment]
I have seen this error before, it's because your metrics is in unnormal range (maybe NaN or INF), which cannot be convert to double. So pls. refine your super parameter, such as 
- ml.learn.rate
- ml.lr.reg.l1
- ml.lr.reg.l2

This iusse can be solved without modify code.

**Comment by qf6101** [comment]
This is the changing trend as I logged. I think there maybe some bug.

```
intercept: -0.019030412734964038, weight0: 1.5686926593335988E-9
intercept: -0.0408969646520407, weight0: 2.882171382998344E-8
intercept: -0.06424865316851307, weight0: -1.5241188168259403E-8
intercept: -0.07030708536934964, weight0: 4.6082353928075444E-8
intercept: -0.0758941345064767, weight0: 8.146194524736786E-8
intercept: -0.08525985684160253, weight0: 1.364594488514541E-7
...
intercept: -Infinite, weight0: NaN
```
